### PR TITLE
chore: pull out get light client module to helper methods

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

Start of #6019 
In order to keep the PRs for this task a little more digestible, I wanted to do this small refactor first. 

And I wanted to ask if I should also use the helper method in other places where we currently do not do any `if !k.GetParams(ctx).IsAllowedClient(clientType)` checks. 

The places I have identified that currently gets the route, but does not run `IsAllowedClient`:
- https://github.com/cosmos/ibc-go/blob/dd23cf9da535d121ef51f4b79f569c025ea22c7a/modules/core/02-client/keeper/keeper.go#L307-L314
- https://github.com/cosmos/ibc-go/blob/dd23cf9da535d121ef51f4b79f569c025ea22c7a/modules/core/02-client/keeper/keeper.go#L395-L411
- https://github.com/cosmos/ibc-go/blob/dd23cf9da535d121ef51f4b79f569c025ea22c7a/modules/core/02-client/keeper/keeper.go#L415-L436
- https://github.com/cosmos/ibc-go/blob/dd23cf9da535d121ef51f4b79f569c025ea22c7a/modules/core/02-client/keeper/client.go#L63-L121

Should they also do this?

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
I was thinking that it might be worth to have feature branch for this, if it's going to span multiple PRs. We have still the benefit of reviewing smaller PRs, but at the same time we also have another opportunity to review how each piece fits together when reviewing the feature branch. What do you think?

**Comment by gjermundgaraba** [comment]
> I was thinking that it might be worth to have feature branch for this, if it's going to span multiple PRs. We have still the benefit of reviewing smaller PRs, but at the same time we also have another opportunity to review how each piece fits together when reviewing the feature branch. What do you think?

Yeah, happy to do that! I will create a feature branch and target that one instead :)

File: modules/core/02-client/keeper/keeper.go [hunk 1]
```diff
@@ -503,3 +476,31 @@ func (k *Keeper) ScheduleIBCSoftwareUpgrade(ctx sdk.Context, plan upgradetypes.P
 
 	return nil
 }
+
+// getLightClientModuleRoute checks that the clientType of clientID is allowed and returns a route to the light client module
+func (k *Keeper) getLightClientModuleRoute(ctx sdk.Context, clientID string) (exported.LightClientModule, error) {
```
Comment (crodriguezvega) [inline modules/core/02-client/keeper/keeper.go:481]:
```suggestion
func (k *Keeper) getLightClientModule(ctx sdk.Context, clientID string) (exported.LightClientModule, error) {
```

**Review by crodriguezvega — COMMENTED** [review]
I'd say that it makes sense to also do the `IsAllowedClient` check in the other functions...

**Review by chatton — APPROVED** [review]
LGTM! 

I also think it makes sense to include the IsAllowed check anywhere we are fetching the client module.

**Pull request merged**